### DISCUSSION TITLE
Hub that works with a RPi that I can buy now.

### DISCUSSION THREAD

**Discussion by xxedgexx** [mvp/uhubctl#518] [General]
I had an old 7 port D-Link DUB-H7 that died.  I'm looking for a replacement.  Tried a couple from Amazon but nothing seems to work as smoothly as the basic D-Link.

What has at least 7 controllable ports that I can purchase right now that's known to work?

Thanks!

**Comment by chuckleu1976** [comment]
Hi, I have designed a kind of hardware, which can switch 16 USB ports totally by ppps.

**Comment by xxedgexx** [comment]
Any more info?

**Comment by chuckleu1976** [comment]
Here are the pictures and chipset 
![微信图片_20230907091215](https://github.com/mvp/uhubctl/assets/124535746/8ee35c5c-d9aa-4945-a2a8-f020a284b717)
![微信图片_20230907091429](https://github.com/mvp/uhubctl/assets/124535746/b5b14f0d-4781-413a-ba9b-2aa48b051f26)
![微信图片_20230907091501](https://github.com/mvp/uhubctl/assets/124535746/75598639-87e3-4f4c-8e0b-6d338348536b)
![微信图片_20230907091508](https://github.com/mvp/uhubctl/assets/124535746/24cd8893-27a2-4696-92d2-e88692012a39)
datasheet

**Comment by chuckleu1976** [comment]
We are working on it, please reference https://github.com/mvp/uhubctl/issues/516

**Comment by xxedgexx** [comment]
Thank you. This sounds very promising.

**Comment by mvp** [comment]
I would recommend UUGear Mega4. It is designed for RPi4B, but works great without it.

If you need higher port count, you can get 2 - Mega4 is stackable on itself.

Also, last time I checked, you can still find silver DLink DUB-H7 on eBay

**Reply by xxedgexx** [reply]
This looks interesting. Can two fit?  It looks like a U shaped usb connector. If I can do two of these, it looks like a good solution.

**Reply by mvp** [reply]
Yes. UUGear Mega4 is specifically designed to be stackable

**Reply by mvp** [reply]
Mega4 developer @uugear specifically created it to fully support uhubctl. He reached out to me before starting designing it. I can attest it works beautifully, with true per port power control. I didn't test stacking it, but documentation says it is supported.

**Reply by xxedgexx** [reply]
Perfect. Ordering two now.

**Reply by xxedgexx** [reply]
Just hooked up two of these to a single pi.  It stacks nicely.  It does require additional power but this is a godsend.  So far working perfectly.  Thank you!
